### PR TITLE
Factions

### PR DESCRIPTION
**Pull request by Jackriip**
## About The Pull Request

This PR adds factions into the code. **There are now three factions - NT-C (Nanotrasen Combat), SYN-C (Syndicate Combat), and NEU (Neutral).**
- NT-C Ships can attack KOS and SYN-C ships. 
- SYN-C can attack KOS and NT-C ships. 
- NEU cannot attack or be attacked by any ship, doing so is breaking the rules.
- You have the option to override any faction tag with KOS. This allows you to fight any ship that has a combat class (NT-C, SYN-C, or KOS).

This also increases the price of the hyena, makes the twinkle able to be purchased, forces a prefix when naming ships, notifies members of their ship's faction on joining, and adds the ship's faction to the purchase list.

Screenshot 2022-01-26 212327

Screenshot 2022-01-26 212402

## Why It's Good For The Game

Makes the gameplay more exciting and gives you greater reason to powergame hard. Helps establish a true gameplay loop. 

Start off as a combat ship and get better gear to win the system-wide arms race. Or play a neutral ship flying under the radar until you're ready to fight, then set your tag to KOS and go raid other combat-factioned ships. 

## Changelog
:cl:
add: Added faction prefixes to all ships
add: Added faction notifications when joining a ship
add: Added factions to ship purchase list
add: Added the twinkle (huge syndicate combat ship) to purchase list
tweak: Automatically attach ship prefix when renaming a ship (if not found)
balance: Upped the hyena cost
/:cl:

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -535,6 +535,25 @@ block( \
 	announcer.announce("ARRIVAL", character.real_name, displayed_rank, list()) //make the list empty to make it announce it in common
 	//WS end
 
+/proc/NotifyFaction(mob/living/carbon/human/character, obj/structure/overmap/ship/simulated/ship)
```
Comment (SuperNovaa41) [inline code/__HELPERS/game.dm:545]: This whole proc looks good, but 
1. It needs some DMDocs.
Essentially just before the proc you would put:
```suggestion
/**
 * short description of the proc
 *
 * Arguments:
 * * mob/living/carbon/human/character - short description of the arg
 * * arg2 - etc
 */
 /proc/NotifyFaction........
```
2. You have return in every single case, along with the else.
Why not take the return out of everyone, remove the else and just have a return outside of the switch statement?

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -730,8 +730,6 @@
 
 	log_game("[key_name(usr)] used abandon mob.")
 
-	to_chat(usr, "<span class='boldnotice'>Please roleplay correctly!</span>")
```
Comment (SuperNovaa41) [inline code/modules/mob/mob.dm:733]: lol

File: code/modules/overmap/helm.dm [hunk 3]
```diff
@@ -194,6 +194,9 @@
 			if(!new_name)
 				return
 			new_name = trim(new_name)
+			var/prefix = current_ship.source_template.prefix
+			if (!(findtext(new_name, "KOS") || findtext(new_name, prefix)))
+				new_name = addtext(prefix, " ", new_name)
```
Comment (SuperNovaa41) [inline code/modules/overmap/helm.dm]: I don't like this, why use addtext? That's an OLD byond proc, that isn't used like anywhere else in the codebase as far as I can tell. Simple string concatenation will work fine here:
```suggestion
				new_name = prefix + " " + new_name
```

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]
The jsons all look okay, also looks like you snuck some line ending changes in here too (energia.dmm + operation_honk.dmm + syndicate_courage.dmm), I would remove those just in case.

**Comment by SuperNovaa41** [comment]
Other than that this is good, this will make creating faction specific access (+ turrets and other small things) way easier.
One thing I would like added is adding their faction to their ID on ship join. If you're not up for that it's ok ill handle it later.

File: code/modules/overmap/helm.dm (hunk 3 above)
Comment (Jackriip) [reply code/modules/overmap/helm.dm]: I used addtext before I knew about being able to use vars directly in strings with "[var]". I can also just append it like that.

**Comment by Jackriip** [comment]
> 

Great idea, that way people know what's what in combat

File: code/modules/overmap/helm.dm (hunk 3 above)
Comment (SuperNovaa41) [reply code/modules/overmap/helm.dm]: yeah `new_name = "[prefix] [new_name]"` would also work

File: code/__HELPERS/game.dm (hunk 1 above)
Comment (Jackriip) [reply code/__HELPERS/game.dm:545]: Should be good to go

**Review by SuperNovaa41 — APPROVED** [review]
Looks good

**Pull request merged**